validateendpoints.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"github.com/getkin/kin-openapi/openapi3"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+var identityToken string
+
+// validateEndpoints tests all paths (represented by openapi3.Paths) with all HTTP methods and given response bodies
+// and make sure they respond with the expected status code. Returns a success bool based on whether all the tests
+// passed.
+func validateEndpoints(paths *openapi3.Paths, identTok string) bool {
+	identityToken = identTok
+
+	success := true
+	for endpoint, pathItem := range *paths {
+		log.Printf("Testing %s endpoint\n", endpoint)
+		success = success && validateEndpointOperation(pathItem.Connect, endpoint, http.MethodConnect)
+		success = success && validateEndpointOperation(pathItem.Delete, endpoint, http.MethodDelete)
+		success = success && validateEndpointOperation(pathItem.Get, endpoint, http.MethodGet)
+		success = success && validateEndpointOperation(pathItem.Head, endpoint, http.MethodHead)
+		success = success && validateEndpointOperation(pathItem.Options, endpoint, http.MethodOptions)
+		success = success && validateEndpointOperation(pathItem.Patch, endpoint, http.MethodPatch)
+		success = success && validateEndpointOperation(pathItem.Post, endpoint, http.MethodPost)
+		success = success && validateEndpointOperation(pathItem.Put, endpoint, http.MethodPut)
+		success = success && validateEndpointOperation(pathItem.Trace, endpoint, http.MethodTrace)
+	}
+
+	return success
+}
+
+// validateEndpointOperation validates a single endpoint and a single HTTP method, and ensures that the request --
+// including the provided sample request body -- elicits the expected status code.
+func validateEndpointOperation(operation *openapi3.Operation, endpoint string, httpMethod string) bool {
+	if operation == nil {
+		return true
+	}
+	log.Printf("%s %s\n", httpMethod, endpoint)
```

Comment:
There are quite a few `logs` sprinkled in here. Are they debugging statements? There doesn't seem to be much context. I would suggest choosing which messages to expose to the end user with care, and use a different logger (or different library) for debugging (so it can be hidden until a user turns it on)